x-pack/plugins/lens/public/indexpattern_datasource/datapanel.tsx

```diff
@@ -503,6 +503,8 @@ export const InnerIndexPatternDataPanel = function InnerIndexPatternDataPanel({
       patterns: [currentIndexPattern.id],
     });
     onUpdateIndexPattern(newlyMappedIndexPattern[currentIndexPattern.id]);
+    // start a new session so all charts are refreshed
+    data.search.session.start();
```

Comment:
You're always refreshing the session, even if the update is not being used within the visualization? I think that it would be better to check the datasource state to see if the field is being used anywhere- although this would require some annoying traversal logic.

I don't have a strong opinion about whether you need to do it in this PR or a followup.